File: src/0374.guess-number-higher-or-lower.374/solution.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):
# 就是二分
class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        low = 1
        high = n
        while True:
            mid = (low+high)//2
            rst = guess(mid)
            if rst == 0 :
                return mid
            elif rst>0 :
                low = mid+1
            else :
                high = mid-1

File: src/0374.guess-number-higher-or-lower.374/solution.java
/**
 * Forward declaration of guess API.
 * 
 * @param num your guess
 * @return -1 if num is higher than the picked number
 *         1 if num is lower than the picked number
 *         otherwise return 0
 *         int guess(int num);
 */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int low = 1;
        int high = n;
        int res = 0;

        while (low <= high) {
            int mid = low + ((high - low) >> 1);
            int compare = guess(mid);
            if (compare == -1) {
                high = mid - 1;
            } else if (compare == 1) {
                low = mid + 1;
            } else {
                res = mid;
                break;
            }

        }
        return res;
    }
}